kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/steps/KafkaSteps.java

```diff
@@ -68,33 +79,35 @@ public static void deleteTopic(String deployNamespace, String topicName, String
-                .withArgs(TOPIC_COMMAND, "delete", BOOTSTRAP_ARG + bootstrap, "--topic=" + topicName)
-                .done();
+        LOGGER.debug("Deleting '{}' topic", topicName);
+        String name = Constants.KAFKA_ADMIN_CLIENT_LABEL + "-delete";
+        List<String> args = Arrays.asList(TOPIC_COMMAND, "delete", BOOTSTRAP_ARG + bootstrap, "--topic=" + topicName);
+        InputStream file = KafkaUtils.replaceStringInResourceFile(ADMIN_CLIENT_TEMPLATE, Map.of(
+                NAMESPACE_VAR, deployNamespace,
+                NAME_VAR, name,
+                ARGS_VAR, String.join(DELIMITER, args),
+                KAFKA_VERSION_VAR, Environment.KAFKA_VERSION));
 
-        DeploymentUtils.waitForPodRunSucceeded(deployNamespace, podName, Duration.ofSeconds(30));
+        kubeClient().getClient().load(file).inNamespace(deployNamespace).create();
+        String podName = KafkaUtils.getPodNameByLabel(deployNamespace, "app", name, Duration.ofSeconds(30));
+        DeploymentUtils.waitForDeploymentRunning(deployNamespace, podName, Duration.ofSeconds(30));
         String log = kubeClient().logsInSpecificNamespace(deployNamespace, podName);
         LOGGER.debug("Admin client delete pod log: {}", log);
     }
 
     private static boolean topicExists(String deployNamespace, String topicName, String bootstrap) {
-        LOGGER.info("Deleting topic {}", topicName);
-        String podName = Constants.KAFKA_ADMIN_CLIENT_LABEL + "-list";
-        kubeClient().getClient().run().inNamespace(deployNamespace).withNewRunConfig()
-                .withImage(Constants.TEST_CLIENTS_IMAGE)
-                .withName(podName)
-                .withRestartPolicy(NEVER_POLICY)
-                .withCommand(ADMIN_CLIENT_COMMAND)
-                .withArgs(TOPIC_COMMAND, "list", BOOTSTRAP_ARG + bootstrap)
-                .done();
+        LOGGER.debug("List '{}' topic", topicName);
+        String name = Constants.KAFKA_ADMIN_CLIENT_LABEL + "-list";
+        List<String> args = Arrays.asList(TOPIC_COMMAND, "list", BOOTSTRAP_ARG + bootstrap);
+        InputStream file = KafkaUtils.replaceStringInResourceFile(ADMIN_CLIENT_TEMPLATE, Map.of(
+                NAMESPACE_VAR, deployNamespace,
+                NAME_VAR, name,
+                ARGS_VAR, String.join(DELIMITER, args),
+                KAFKA_VERSION_VAR, Environment.KAFKA_VERSION));
 
-        DeploymentUtils.waitForPodRunSucceeded(deployNamespace, podName, Duration.ofSeconds(30));
+        kubeClient().getClient().load(file).inNamespace(deployNamespace).create();
+        String podName = KafkaUtils.getPodNameByLabel(deployNamespace, "app", name, Duration.ofSeconds(30));
+        DeploymentUtils.waitForDeploymentRunning(deployNamespace, podName, Duration.ofSeconds(30));
         String log = kubeClient().logsInSpecificNamespace(deployNamespace, podName);
         LOGGER.debug("Admin client list pod log: {}", log);
         return log.contains(topicName);
```

Comment:
this test is really weak.   It will give the wrong answer for topics name that are substrings of of other topics `foo`, `foobar` etc.  This will lead to incorrect behaviour.

Reply:
I went to this solution because there is no flag supported, but `-all` in our [test clients](https://github.com/strimzi/test-clients/blob/main/admin/src/main/java/io/strimzi/arguments/topic/DeleteTopicCommand.java), and because we always use a random names for topics and it is really difficult to have two similar names at the same time in a test case.

Reply:
> because we always use a random names for topics and it is really difficult to have two similar names at the same time in a test case.

No, that's not the right approach.  There's nothing in the API  (contract) of these methods that insists the caller uses random names so the caller is within their rights to use 'foo' or 'foobar'    Writing code that makes assumptions about how it will be used it really bad practice.  (And it always bites you - or someone else - in the end).

Let's write the best code we can - that needs to apply equally to production code and test code.

Reply:
Solved, using a similar approach, but getting the names into a list to check the object, so any discrepancy should be raised.

Reply:
BTW, I have created an issue to request to add that useful option to delete a topic and remove our checking when it is included in the test-clients
https://github.com/strimzi/test-clients/issues/83

Reply:
Thank you, that's is indeed a safer approach, which is good enough.